--- ../original/10-tmpdirs	2011-04-25 18:48:26 -0500
+++ ./10-tmpdirs	2011-04-26 01:00:36 -0500
@@ -30,7 +30,7 @@
 for dir in /tmp/.ICE-unix /tmp/.X11-unix /tmp/.font-unix ; do
 	# Use mktemp rather than mkdir to avoid possible security issue
 	# if $dir exists and is a symlink
-	if mktemp -d ${dir} >& /dev/null ; then
+	if /usr/bin/mktemp -d ${dir} >& /dev/null ; then
 		chmod 1777 $dir
 		chown root:wheel $dir
 	fi
